test/react/navbar.test.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import MyNavbar from "../../app/javascript/components/my_navbar";
+import { expect, test } from "@jest/globals";
+import { getByTestId, queryByText, getByText, render, RenderResult } from "@testing-library/react";
+import "@testing-library/jest-dom/extend-expect";
+import "@testing-library/jest-dom";
+import "jest-extended";
+
+const renderNavbar = (
+  loggedin = true,
+  regPath = "/signup",
+  homePath = "/"
+) : RenderResult => {
+  return render(
+    <MyNavbar logged_in= {loggedin} regPath={regPath} homePath={homePath} />
+  );
+};
+
+test("Component renders as expected", () => {
+  const { container } = renderNavbar();
+  expect(container).toMatchSnapshot();
+});
+
+test("Title button has been passed the right link", () => {
```

Comment:
When writing React tests, we would usually use a functional testing approach.
This would mean that we're not concerned with how the component achieves what we're testing, but instead focuses on the output. 

In that case, the test case should be something like `Home button has the correct link`.

If you want to read more about functional testing, this is a pretty good article:
https://medium.com/javascript-scene/behavior-driven-development-bdd-and-functional-testing-62084ad7f1f2